docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,652 @@
+Congratulations! You now have a working FastAPI application with routes to add, list, and delete items from a grocery list.
+
+## Set up the data storage
+At this point, you already have a working version of the application with the base functionality. This section will guide you through setting up data storage for persistence, but you can choose to skip it if you're happy with what you've learned already.
+
+So far we are storing the data in a dictionary, which is not ideal because all of the data will be lost when the application is restarted.
+
+To persist the data, we will use [Redis](https://redis.io/), which is an open source in-memory data structure store. Due to its speed and versatility, Redis is commonly used as a data storage system in a wide range of applications, including web applications, real-time analytics systems, caching layers, this tutorial, and more.
+
+If you are already working on **GitHub Codespaces** with our existing template, you can skip directly to the ["Replace the database"](#replace-the-database) section.
+
+If you are on Windows, you can work with Redis by setting up either a [Docker container](https://www.docker.com/products/docker-desktop/), a [GitHub Codespace](https://github.com/features/codespaces) or through [WSL](https://learn.microsoft.com/en-us/windows/wsl/) (Windows Subsystem for Linux). In this tutorial we will use a Docker container, but you can refer to the [section above](#github-codespaces) for instructions on how to set up a GitHub Codespace.
+
+Otherwise, if you are on a Linux or a macOS machine, you can install Redis by following the [instructions on their website](https://redis.io/docs/getting-started/), and then skip to the ["Replace the database"](#replace-the-database) section.
+
+### Setting up a Docker Container on Windows
+The Dev Container extension for VS Code offers a simple way to configure a Linux-based workspace in your Windows machine, putting your entire project and its dependencies as well as tools you need for your project into a tidy container.
+For the steps below, make sure you have the following requirements installed on your machine:
+
+#### Requirements
+-	[Docker for Windows](https://www.docker.com/)
+-	[Dev Containers extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+
+#### Create the Dev Container configuration
+1. Open the Command Palette and run the “Dev Container: Add Dev Container Configuration Files…"
+2. Select "Python 3":
+
+    <img src="images/fastapi-tutorial/devcontainers_python3.png" alt="Python 3 option selected in the Dev Container configuration files list." width="500"/>
+3. Select the default version.
+
+We can optionally install features to be included in the container. For this tutorial, we will install [Redis Server](https://github.com/itsmechlark/features/tree/main/src/redis-server), which is a community contributed feature that installs and adds the proper dev container set up for Redis.
```

Comment:
"Features" should be capitalized here and throughout the rest of the doc. Would also recommend providing some more info on what Features are and linking to their docs (as I'd expect many users aren't familiar with them)

```suggestion
We can optionally install Features to be included in the container. For this tutorial, we will install [Redis Server](https://github.com/itsmechlark/features/tree/main/src/redis-server), which is a community contributed Feature that installs and adds the proper dev container set up for Redis.
```